src/status_im/contexts/wallet/wallet_connect/modals/common/footer/view.cljs

```diff
@@ -6,20 +6,44 @@
             [status-im.common.standard-authentication.core :as standard-authentication]
             [status-im.contexts.wallet.sheets.buy-token.view :as buy-token]
             [status-im.contexts.wallet.wallet-connect.modals.common.footer.style :as style]
+            [utils.hex :as hex]
             [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
 (defn- on-auth-success
-  [password]
-  (rf/dispatch [:hide-bottom-sheet])
-  (rf/dispatch [:wallet-connect/authorized-signing password]))
+  [signatures]
+  (let [signature (-> signatures
+                      first
+                      :signature
+                      hex/prefix-hex)]
+    (rf/dispatch [:hide-bottom-sheet])
+    (rf/dispatch [:wallet-connect/respond signature])))
+
+(defn- on-auth-fail
+  [error]
+  (rf/dispatch [:wallet-connect/on-sign-error error]))
+
+(defn- slide-button
+  [{:keys [disabled? slide-button-text]}]
+  (let [{:keys [customization-color]} (rf/sub [:wallet-connect/current-request-account-details])
+        address                       (rf/sub [:wallet-connect/current-request-address])
+        prepared-hash                 (rf/sub [:wallet-connect/prepared-hash])]
+    [rn/view {:style style/auth-container}
+     [standard-authentication/slide-button
+      {:sign-payload        [{:address address
+                              :message prepared-hash}]
+       :size                :size-48
+       :track-text          slide-button-text
+       :disabled?           (and (not prepared-hash) disabled?)
+       :customization-color customization-color
+       :on-auth-success     on-auth-success
```

Comment:
if we have sign-payload then there will be on-sign-success, so probably it should be just on-success ? without auth, so it will work for auth and sign

Reply:
idk actually , so just by adding `sign-payload` it's not just standard-authentication anymore, we now have signing mechanism inside standard-authentication, little bit confusing

Reply:
so probably it's better to have separate component wrapper, like signing/slide-button which will wrap standard-authentication/slide-button

Reply:
agreed, given that the slide button was stripped of logic, now it's easier to have two separate ones. Thanks!